src/main/java/emu/grasscutter/game/player/Player.java

```diff
@@ -797,6 +800,43 @@ public void addAvatar(int avatarId) {
         addAvatar(new Avatar(avatarId), true);
     }
 
+    public boolean addTrialAvatarForQuest(int trialAvatarId, int questMainId){
+        // TODO, other trial avatar like activity and element trial dungeon might have 
+        // completely different scenario, this function is currently used for Quest Exec only
+        TrialAvatarData trialAvatar = GameData.getTrialAvatarDataMap().get(trialAvatarId);
+        Avatar avatar = new Avatar(trialAvatar.getTrialAvatarParamList().get(0));
+        if (avatar.getAvatarData() == null || !hasSentLoginPackets()) {
+            return false;
+        }
+        avatar.setOwner(this);
+        // Add trial weapons and relics
+        avatar.setTrialAvatarInfo(trialAvatar, GrantReason.GRANT_REASON_BY_QUEST, questMainId);
```

Comment:
would probably be good to have the GrantReason as parameter, since in the long term there will be many sources for them.